src/main/java/org/zephyrsoft/sdb2/gui/MainWindow.java

```diff
@@ -482,7 +483,18 @@ public void setModels(SongsModel songs, SettingsModel settings) {
 		songsListModel = songs.getFilterableListModel();
 		songsList.setModel(songsListModel);
 		
-		songsModel.addSongsModelListener(() -> indexAllSongs());
+		// If some song changes, index changes, update the list and load the current song again
+		songsModel.addSongsModelListener(() -> {
+			indexAllSongs();
+			
+			FieldName[] fieldsToSearch = settingsModel.get(SettingKey.SONG_LIST_FILTER, FilterTypeEnum.class).getFields();
+			songsListFiltered = indexer.search(IndexType.ALL_SONGS, textFieldFilter.getText(), fieldsToSearch);
+			songsListModel.refilter();
+			// TODO: implement a merging strategy, by comparing changes with the current opened song.
```

Comment:
What about this? Seems like there's still something to do...

Reply:
ok. Added my todos as issues to https://github.com/l-spiecker/sdb2/issues with commit https://github.com/l-spiecker/sdb2/commit/52f048a3395a03851b9ed6588e16c67b8123fb1e.

There are still 27 todos left in sdb2.

Reply:
:+1: Thanks! You're right, there are way too many TODOs in the code base, bet let's at least keep them from growing even more.